pkg/storer/storer.go

```diff
@@ -278,15 +278,15 @@ func initDiskRepository(
 	basePath string,
 	locker storage.ChunkLocker,
 	opts *Options,
-) (storage.Repository, io.Closer, error) {
+) (storage.Repository, *PinIntegrity, io.Closer, error) {
```

Comment:
instead of returning so many things from the constructor, the PinIntegrity can returned with a get function?

Reply:
so I tried adding a getter to the `storage.Repository` but the compiler says that now there are `183` places that need changing (to comply with the new interface) so it seems it's no feasible. I'll look into other ways but for now I don't have a better approach.